### PR TITLE
Feature/metacoins

### PR DESCRIPTION
**Pull request by Duelist**

### CODE REVIEW

File: commands/metacoins/constants.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// COMMANDS //
+//////////////
+
+
+
+/**
+ * Command for awarding metacoins.
+ * @type {String}
+ */
+const AWARD = 'award'
```
Comment (Duelist) [inline commands/metacoins/constants.js]: 💣

File: commands/metacoins/constants.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  LEADERBOARD
+}
+
+
+
+/**
+ * Message for an awarded metacoin.
+ * @type {String}
+ */
+const AWARDED_MESSAGE = template`A metacoin has been awarded to ${0}.`
```
Comment (Duelist) [inline commands/metacoins/constants.js]: 💣

File: commands/metacoins/constants.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+
+
+/**
+ * User ID used to denote the admin.
+ * @type {String}
+ */
+const ADMIN_USER_ID = '179139616274644992'
```
Comment (Duelist) [inline commands/metacoins/constants.js]: Use env var.

File: constants/metacoins.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+// REDIS //
+///////////
+
+
+
+/**
+ * Redis key for last award times for registered functions.
+ * @type {String}
+ */
+const LAST_AWARDED_REDIS_KEY = 'lastAwarded'
```
Comment (Duelist) [inline constants/metacoins.js]: `REDIS_LAST_AWARDED_KEY`

File: constants/metacoins.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * Redis key for metacoin leaderboard.
+ * @type {String}
+ */
+const LEADERBOARD_REDIS_KEY = 'leaderboard'
```
Comment (Duelist) [inline constants/metacoins.js]: `REDIS_LEADERBOARD_KEY`

File: test/utils/test.js [hunk 6]
```diff
@@ -29,3 +29,14 @@ describe('#randomString', () => {
...
 
 })
+
+
+
+describe('#rewireRoot', () => {
+
+  it('rewires a module at a given path from root', () => {
+    let result = testUtil.rewireRoot('utils/test')
+    true.should.eql(typeof result === 'object')
```
Comment (Duelist) [inline test/utils/test.js]: 📝

File: commands/metacoins/index.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * @param {Object} options
+ */
+function* message(options) {
+
+  if (!options.args || R.isEmpty(options.args)) {
+    let coins = yield metacoins.getMetacoinsForUser(
+      options.message.author.id
+    )
+    yield options.message.channel.sendMessage(coins.toString())
```
Comment (Duelist) [inline commands/metacoins/index.js]: Use a better message.

**Comment by Fikaf** [comment]
i think it looks aight, although i find it annoying that constants arent all defined in a common file somewhere, or on the top of the file.
╱╱┏╮
╱╱┃┃
▉━╯┗━╮
▉┈┈┈┈┃
▉╮┈┈┈┃
╱╰━━━╯

**Comment by Duelist** [comment]
Constants could be used multiple times throughout different parts of the code, and are separated based on their domain.

**Pull request merged**